modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medguns.dm

```diff
@@ -0,0 +1,169 @@
+	w_class = WEIGHT_CLASS_SMALL
+	icon_state = "Brute1"
+	ammo_type = /obj/item/ammo_casing/energy/medical/brute2
+//Burn II//
+/obj/item/medicell/burn2
+	name = "Burn II Medicell"
+	desc = "A small cell with a intense yellow glow. Can be used on Mediguns to unlock the Burn II Functoinality"
+	w_class = WEIGHT_CLASS_SMALL
+	icon_state = "Burn1"
+	ammo_type = /obj/item/ammo_casing/energy/medical/burn2
+//Toxin II//
+/obj/item/medicell/toxin2
+	name = "Toxin II Medicell"
+	desc = "A small cell with a intense green glow. Can be used on Mediguns to unlock the Toxin II Functoinality"
+	w_class = WEIGHT_CLASS_SMALL
+	icon_state = "Toxin1"
+	ammo_type = /obj/item/ammo_casing/energy/medical/toxin2
+//Oxygen II//
+/obj/item/medicell/oxy2
+	name = "Oxygen II Medicell"
+	desc = "A small cell with a intense blue glow. Can be used on Mediguns to unlock the Oxygen II Functoinality"
+	w_class = WEIGHT_CLASS_SMALL
+	icon_state = "Oxygen1"
+	ammo_type = /obj/item/ammo_casing/energy/medical/oxy2
+//End of Tier II
+//Tier III Cells/
+//Brute III//
+/obj/item/medicell/brute3
+	name = "Brute III Medicell"
+	desc = "A small cell with a intense red glow. Can be used on Mediguns to unlock the Brute II Functoinality"
+	w_class = WEIGHT_CLASS_SMALL
+	icon_state = "Brute1"
+	ammo_type = /obj/item/ammo_casing/energy/medical/brute3
+//Burn III//
+/obj/item/medicell/burn3
+	name = "Burn III Medicell"
+	desc = "A small cell with a intense yellow glow. Can be used on Mediguns to unlock the Burn II Functoinality"
+	w_class = WEIGHT_CLASS_SMALL
+	icon_state = "Burn1"
+	ammo_type = /obj/item/ammo_casing/energy/medical/burn3
+//Toxin III//
+/obj/item/medicell/toxin3
+	name = "Toxin III Medicell"
+	desc = "A small cell with a intense green glow. Can be used on Mediguns to unlock the Toxin II Functoinality"
+	w_class = WEIGHT_CLASS_SMALL
+	icon_state = "Toxin1"
+	ammo_type = /obj/item/ammo_casing/energy/medical/toxin3
+//Oxygen III//
+/obj/item/medicell/oxy3
+	name = "Oxygen III Medicell"
+	desc = "A small cell with a intense blue glow. Can be used on Mediguns to unlock the Oxygen II Functoinality"
+	w_class = WEIGHT_CLASS_SMALL
+	icon_state = "Oxygen1"
+	ammo_type = /obj/item/ammo_casing/energy/medical/oxy3
+//End of Tier III
+//Medigun Upgrade//
+/obj/item/gun/energy/medigun/attackby(obj/item/medicell/M, mob/user)
+	if(cellcount >= maxcells)
+		to_chat(user, span_notice("The Medigun is full, take a cell out to make room"))
+	else
+		if(!user.transferItemToLoc(M, src))
+			return
+		playsound(loc, 'sound/machines/click.ogg', 50, 1)
+		to_chat(user, span_notice("You install the medicell."))
+		ammo_type += new M.ammo_type(src)
+		installedcells += M
+		cellcount += 1
+
+/obj/item/gun/energy/medigun/RightClick(mob/user)
+	if(cellcount >= 1)
+		to_chat(user, span_notice("You remove a cell"))
+		var/obj/item/last_cell = installedcells[installedcells.len]
+		if(last_cell)
+			last_cell.forceMove(drop_location())
+		installedcells -= last_cell
+		installedcells.len--
```

Comment:
```suggestion
```
Didn't see this line; thats my fault

Reply:
ah, it's good. Thank you for reviewing my code by the way!